opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -141,18 +146,41 @@ namespace Opm
                 if (numSamples > 0) {
                     auto splinePvt = std::shared_ptr<PvtDeadSpline>(new PvtDeadSpline);
                     splinePvt->initFromOil(pvdoTables, numSamples);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        splinePvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                     deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = splinePvt;
                 } else {
                     auto deadPvt = std::shared_ptr<PvtDead>(new PvtDead);
                     deadPvt->initFromOil(pvdoTables);
+
+                    if (!eclState->getOilvisctTables().empty()) {
+                        deadPvt->setOilvisctTables(eclState->getOilvisctTables(),
+                                                   deck->getKeyword("VISCREF"));
+                    }
+
                     props_[phase_usage_.phase_pos[Liquid]] = deadPvt;
                 }
             } else if (!pvtoTables.empty()) {
-                props_[phase_usage_.phase_pos[Liquid]].reset(new PvtLiveOil(pvtoTables));
```

Comment:
i do not understand why this change (though minor) has been made. I see the handling is not uniform between pvto and pvcdo below, so if you want to make them uniform that is ok. In that case, use pvto not pvt as the variable name and use a shared_ptr.